src/sequence_manager.cpp

```diff
@@ -44,25 +46,50 @@ bool SequenceManager::sequenceExists(const uint64_t sequenceId) const {
     return sequences.count(sequenceId);
 }
 
-Status SequenceManager::removeTimedOutSequences(std::chrono::steady_clock::time_point currentTime) {
+Status SequenceManager::removeTimedOutSequences() {
+    std::unique_lock<std::mutex> sequenceManagerLock(mutex);
     for (auto it = sequences.cbegin(); it != sequences.cend();) {
-        auto& sequence = it->second;
-        auto timeDiff = currentTime - sequence.getLastActivityTime();
-        if (std::chrono::duration_cast<std::chrono::seconds>(timeDiff).count() > timeout)
+        Sequence& seqRef = getSequence(it->second.getId());
```

Comment:
```suggestion
        Sequence& sequence = getSequence(it->second.getId());
```
I think `sequence` is enough. I think we use it everywhere else in the code instead of `seqRef `or `sequenceRef`.